src/main/java/com/redhat/rhjmc/containerjfr/platform/internal/KubeApiPlatformClient.java

```diff
@@ -100,15 +107,16 @@
             return Collections.emptyList();
         }
     }
-
+    /*
     private ServiceRef resolveServiceRefHostname(ServiceRef in) {
         try {
-            String hostname = resolver.resolveCanonicalHostName(in.getConnectUrl());
-            logger.debug(String.format("Resolved %s to %s", in.getConnectUrl(), hostname));
-            return new ServiceRef(hostname, in.getPort());
+            String hostname = resolver.resolveCanonicalHostName(in.getJMXServiceUrl());
+            logger.debug(String.format("Resolved %s to %s", in.getJMXServiceUrl(), hostname));
+            return new ServiceRef(hostname);
         } catch (Exception e) {
             logger.debug(e);
             return null;
         }
     }
+    */
```

Comment:
Is this intended to be removed? The point of this method was to map IP addresses to more readable hostnames for the entry aliases, since the IP address itself isn't likely to be meaningful to a human reader without them having to go look at their cluster configuration and cross-reference the addresses. A hostname is more likely to give some telling information about what the user is looking at.

Reply:
Yeah no, I couldn't figure it out so I had temprarily removed it. Is the resolveCanonicalHostName only important for the alias? Because I could just incorporate it when Im creating the serviceRef above, and insert the resolved hostname as alias?

Reply:
Yes, that's all it does is use a reverse DNS lookup to turn an InetAddress into a hostname string (under the hood it's just https://docs.oracle.com/javase/7/docs/api/java/net/InetAddress.html#getCanonicalHostName()), ex. `127.0.0.1 -> localhost.local` or `123.45.6.7 -> someProductionService.cluster.local`.

Using the actual IP to construct the JMXServiceURL makes sense, since if that URL used the hostname then it would need to do another DNS lookup back in the other direction to resolve the IP anyway (whoops - that's how it used to work, re-reading this). But that reverse-lookup-resolved hostname should be kept around for use as the alias as I said above.

What part did you have trouble figuring out? Was it the stream mapping that was also commented out above?

Reply:
I tried reducing it given that there was no point in creating the serviceRef and then making a new one again, im not sure if I should add back in the log message?

Reply:
I think it's fine to remove it.